addons/l10n_es_edi_sii/models/account_edi_format.py

```diff
@@ -0,0 +1,587 @@
+        if invoices[0].is_sale_document():
+            return {'url': 'https://www2.agenciatributaria.gob.es/static_files/common/internet/dep/aplicaciones/es/aeat/ssii_1_1/fact/ws/SuministroFactEmitidas.wsdl'}
+        else:
+            return {'url': 'https://www2.agenciatributaria.gob.es/static_files/common/internet/dep/aplicaciones/es/aeat/ssii_1_1/fact/ws/SuministroFactRecibidas.wsdl'}
+
+    def _l10n_es_edi_web_service_bizkaia_vals(self, invoices):
+        if invoices[0].is_sale_document():
+            return {
+                'url': 'https://www.bizkaia.eus/ogasuna/sii/documentos/SuministroFactEmitidas.wsdl',
+                'test_url': 'https://pruapps.bizkaia.eus/SSII-FACT/ws/fe/SiiFactFEV1SOAP',
+            }
+        else:
+            return {
+                'url': 'https://www.bizkaia.eus/ogasuna/sii/documentos/SuministroFactRecibidas.wsdl',
+                'test_url': 'https://pruapps.bizkaia.eus/SSII-FACT/ws/fr/SiiFactFRV1SOAP',
+            }
+
+    def _l10n_es_edi_web_service_gipuzkoa_vals(self, invoices):
+        if invoices[0].is_sale_document():
+            return {
+                'url': 'https://egoitza.gipuzkoa.eus/ogasuna/sii/ficheros/v1.1/SuministroFactEmitidas.wsdl',
+                'test_url': 'https://sii-prep.egoitza.gipuzkoa.eus/JBS/HACI/SSII-FACT/ws/fe/SiiFactFEV1SOAP',
+            }
+        else:
+            return {
+                'url': 'https://egoitza.gipuzkoa.eus/ogasuna/sii/ficheros/v1.1/SuministroFactRecibidas.wsdl',
+                'test_url': 'https://sii-prep.egoitza.gipuzkoa.eus/JBS/HACI/SSII-FACT/ws/fr/SiiFactFRV1SOAP',
+            }
+
+    def _l10n_es_edi_call_web_service_sign(self, invoices, info_list):
+        company = invoices.company_id
+        # All are sharing the same value, see '_get_batch_key'.
+        csv_number = invoices.mapped('l10n_es_edi_csv')[0]
+
+        # === Decode the certificate ===
+
+        certificate = invoices.company_id.l10n_es_edi_certificate_id
+        pem_certificate, pem_private_key, certificate = certificate._decode_certificate()
+        with NamedTemporaryFile(suffix='.cert') as cert_file, NamedTemporaryFile(suffix='.pem') as pem_file:
+            cert_file.write(pem_certificate)
+            cert_file.flush()
+            pem_file.write(pem_private_key)
+            pem_file.flush()
+
+            # === Call the web service ===
+
+            class L10nEsHTTPAdapter(requests.adapters.HTTPAdapter):
+                def init_poolmanager(self, *args, **kwargs):
+                    kwargs['ssl_context'] = create_urllib3_context(ciphers=EUSKADI_CIPHERS)
+                    return super().init_poolmanager(*args, **kwargs)
+
+            class ESTransport(zeep.transports.Transport):
+                def __init__(self, *args, **kwargs):
+                    # OVERRIDE: block DH ciphers for Euskadi servers.
+                    super().__init__(*args, **kwargs)
+                    self.session.mount('https://', L10nEsHTTPAdapter())
```

Comment:
Why are you creating classes inline during a function call? (assuming that's even useful)

Reply:
@smetl For beauty of code I guess